src/wp-includes/block-supports/border.php

```diff
@@ -135,6 +135,37 @@ function wp_skip_border_serialization( $block_type ) {
 		$border_support['__experimentalSkipSerialization'];
 }
 
+/**
+ * Checks whether the current block type supports the border feature requested.
+ *
+ * If the `__experimentalBorder` support flag is a boolean `true` all border
+ * support features are available. Otherwise, the specific feature's support
+ * flag nested under `experimentalBorder` must be enabled for the feature
+ * to be opted into.
+ *
+ * @since 5.8.0
+ * @access private
+ *
+ * @param WP_Block_Type $block_type Block type to check for support.
+ * @param string        $feature    Name of the feature to check support for.
+ * @param mixed         $default    Fallback value for feature support, defaults to false.
+ *
+ * @return boolean Whether or not the feature is supported.
+ */
+function wp_has_border_feature_support( $block_type, $feature, $default = false ) {
+	// Check if all border support features have been opted into via `"__experimentalBorder": true`.
+	if (
+		property_exists( $block_type, 'supports' ) &&
+		( true === _wp_array_get( $block_type->supports, array( '__experimentalBorder' ), $default ) )
+	) {
+		return true;
+	}
```

Comment:
This part can probably use `block_has_support` as well without the nested `$feature` thing but I guess that comment is better in the Gutenberg repo :)

Reply:
Yes, let's fix in Gutenberg and backport with 10.7 version.